src/components/core/MultiStepButton.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { FC, useState, useCallback } from 'react';
+import styled from 'styled-components';
+
+import { BubbleButton, UnstyledButton } from './Button';
+
+enum MultiStepState {
+  DEFAULT,
+  APPROVE,
+}
+
+const { APPROVE, DEFAULT } = MultiStepState;
+
+interface Props {
+  className?: string;
+  needsUnlock?: boolean;
+  valid: boolean;
```

Comment:
You could use `spender` and `token` for props, and then use an allowance hook in the component; make it self contained

Reply:
Skipping this for now as I'd need to pass down the the input value too